src/parquet/encodings/plain-encoding.h

```diff
@@ -71,6 +71,22 @@ inline int PlainDecoder<parquet::Type::BYTE_ARRAY>::Decode(ByteArray* buffer,
   return max_values;
 }
 
+// Template specialization for FIXED_LEN_BYTE_ARRAY
+template <>
+inline int PlainDecoder<parquet::Type::FIXED_LEN_BYTE_ARRAY>::Decode(FixedLenByteArray* buffer,
```

Comment:
How about dictionary encoding (see the `BYTE_ARRAY` specialization there)? It's hard to verify that it works without tests, though, we should try to address that soon.